File: day1.erl
-module(day1).

-export([part1/1, part2/1]).

part1(Lines) ->
    check_heights(["999"] ++ [list_to_integer(Line) || Line <- Lines], 0).


check_heights([_], Acc) -> Acc;
check_heights([H1, H2 | T], Acc) when H1 < H2 ->
    check_heights([H2|T], Acc + 1);
check_heights([_|T], Acc) ->
    check_heights(T, Acc).

check_heights_window([_, _, _], Acc) -> Acc;
check_heights_window([H1, H2, H3, H4 | T], Acc) when H1 < H4 ->
    check_heights_window([H2, H3, H4|T], Acc + 1);
check_heights_window([_|T], Acc) ->
    check_heights_window(T, Acc).

part2(Lines) ->
    check_heights_window([list_to_integer(Line) || Line <- Lines], 0).

File: Eva's Garbage Fire/day1.py
print('hello world')


def part1():
    file = open('day1.txt', 'r')

    counter = 0
    previous_number = 10000

    for line in file:
        number = int(line)

        if number > previous_number:
            counter = counter+1

        previous_number = number

    print(counter)
    file.close()


part1()


def part1m(offset: int, lines):
    return sum([1 for index, line in enumerate(lines[:-offset]) if int(line) < int(lines[index + offset])])


def part2():
    file = open('day1.txt', 'r')

    counter = 0
    large_number = 1000000

    previous_previous_previous_number = 1000000
    previous_previous_number = 1000000
    previous_number = 1000000

    for line in file:
        number = int(line)

        if number+previous_number+previous_previous_number > previous_number+previous_previous_number+previous_previous_previous_number:
            counter = counter+1

        previous_previous_previous_number = previous_previous_number
        previous_previous_number = previous_number
        previous_number = number

    print(counter)
    file.close()

lines = open('day1.txt', 'r').readlines()
print(part1m(1, lines))
print(part1m(3, lines))





